### DISCUSSION TITLE
Missing option to enable prebuild for all branches

### DISCUSSION THREAD

**Discussion by AjXUthaya** [community/community#46402] [Codespaces] [answered]
### Select Topic Area

Product Feedback

### Body

One of the options i am missing from my experience gitpod is the option to enable prebuild for all my branches instead of having to set it up for every branch manually. This enables you to boot up a new codespace relatively fast for new branches as well, lets say for a code review. Are there any plans to add this feature? Right now i work on feature branches and have to enable prebuild on every one of them or simply not use the fast boot up feature prebuild offers when opening a new codespace.

**Comment by chris-c-thomas** [comment, marked as answer]
According to these [docs here](https://docs.github.com/en/codespaces/prebuilding-your-codespaces/configuring-prebuilds#configuring-prebuilds):

<img width="760" alt="Screenshot 2023-02-05 at 12 12 49 AM" src="https://user-images.githubusercontent.com/22353987/216804689-1c01c653-9b97-4bc0-a6fc-27e80dc8f4a5.png">

So I think if your set the prebuilds to the branch your feature branches are based off of they should all be identical.

**Comment by bjbennet** [comment]
Hi, @AjXUthaya as mentioned in the comment above, if you've enabled prebuilds on a branch that your feature branches are based on, when creating a new codespace on those feature branches you will get a prebuild given the dev container configurations is the same as the base branch. This way you don't have to enable prebuilds on each feature branch.

**Comment by casperdcl** [comment]
What about if the devcontainer config is different?

[Use case](https://github.com/microsoft/vscode-remote-release/issues/6989#issuecomment-2985710175):

> a devcontainer docker image which optionally uses a GPU if detected.

Right now `"gpu": "optional"` does not auto-append `"--gpus=all"` to the `"runArgs"` if a GPU Machine type is selected in GH Codespaces. So there has to be 2 separate branches with different config.

But said branches need to be on 2 different repos to prebuild both 😬.
